src/withHydrationSuppress.tsx

```diff
@@ -27,8 +32,10 @@ export const withHydrationSuppress = function <T>(
     return Component ? (
       <Component {...props} />
     ) : (
-      <RootElement
-        dangerouslySetInnerHTML={{ __html: "" }}
+      <div
+        dangerouslySetInnerHTML={{
+          __html: element ?? "",
+        }}
```

Comment:
I'm not sure this will work 🤔. `dangerouslySetInnerHTML` only set innerHTML, so you'll still have a mismatching element.

From what @merykozlowska and I observed, if the root of the element has a different tag from the rendered element with `dangerouslySetInnerHTML` (in this case a `div`), then we will observe the bug (the element will get removed from the DOM).
If the tag match, then `dangerouslySetInnerHTML={{ __html: "" }}` will work as expected (meaning the element will be left untouched in the DOM). No need to actually set it to the HTML of the element. Yes that's really not intuitive 🤷‍♂️.

I do agree that knowing the tag of the root element of the MFE you're loading can be a bit constraining though.
Another way would be to something similar to what you're doing now and pass the `id` of the element, then use that to find the element and use the correct tag. This does require all MFE to let you set the id of their root though (or having an `id`), which is another constraint 🤷‍♂️.

Personally I feel telling the tag of the root is less constraining than having an id (also the code to handle it is easier) but in both cases it's knowledge of the loaded component so it's not ideal.

Reply:
If that's what `__html: ""` is doing, then I'm fine with the current code 👍 

I just feel like there should be an easier way to mount a MFE with the current implementation. 
suggestion: Maybe we could have the root element type exposed by contract so that consumers don't have to figure out which kind of element they need to mount?

On the other hand, the approach I wrote would only copy the previous rendered html and paste it temporarily until
it is able to load the component from the passed `loader`.

Reply:
> the approach I wrote would only copy the previous rendered html and paste it temporarily until
it is able to load the component from the passed loader.

I'm not sure I understand that bit, I might be wrong but I think the code you're offering would do the following.

Given that the component when rendered do:
```html
<p>Hello!</p>
```

The client side "while we wait" html would be:
```html
<div>
<p>Hello!</p>
</div>
```

This is because `<div dangerouslySetInnerHTML={{  __html: element }}` always renders as a `div` with whatever is passed to `dangerouslySetInnerHTML` inside.
The element at the root (`div`) would mismatch with the DOM (`p`) and React would remove everything (like it does now with the bug).

Also we probably want to avoid passing arbitrary HTML to `dangerouslySetInnerHTML` to avoid XSS :)

Reply:
I guess you are right, it would inject inside a div.
Sounds good, lets leave it as is then! I'll close the PR.

Reply:
my understanding of how it works is the same as Adrien's - so i believe this solution with `id` won't work.

i agree that it's not great to need to know that root element of the loaded component. but with `id` you need to know the `id`, so it doesn't really solve this problem 🤷‍♀️ 

> Maybe we could have the root element type exposed by contract so that consumers don't have to figure out which kind of element they need to mount?

that's a good idea. if it's exposed in the contract it's much less of a problem that the consumer has to know it 👍